File: src/main/java/com/scottscmo/ui/container/CommandRunner.java
package com.scottscmo.ui.container;

import java.awt.Color;
import java.util.List;

import javax.swing.JButton;
import javax.swing.JComponent;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JTextArea;

import net.miginfocom.swing.MigLayout;

public class CommandRunner extends JPanel {

    public CommandRunner() {
        super();

        // components init

        JLabel inputLabel = new JLabel("Input:");
        JTextArea commandInput = new JTextArea();
        JButton runBtn = new JButton("Run");
        JLabel outputLabel = new JLabel("Output:");
        JTextArea outputDisplay = new JTextArea();

        // components config

        commandInput.setRows(3);

        outputDisplay.setEditable(false);

        runBtn.addActionListener(actionEvent -> {
            try {
                runCommand(commandInput.getText());
            } catch (Exception e) {
                outputDisplay.setForeground(Color.RED);
                outputDisplay.setText(e.getMessage());
            }
        });

        // layout

        this.setLayout(new MigLayout());
        List<JComponent> components = List.of(
            inputLabel,
            commandInput,
            runBtn,
            outputDisplay,
            outputLabel
        );
        components.forEach(cmp -> this.add(cmp, "wrap"));
    }

    private void runCommand(String command) {

    }
}

File: src/main/kotlin/com/scottscmo/ui/container/CommandRunner.kt
package com.scottscmo.ui.container

import net.miginfocom.swing.MigLayout
import java.awt.Color
import javax.swing.JPanel
import javax.swing.JLabel
import javax.swing.JTextArea
import javax.swing.JButton
import java.lang.Exception
import java.util.function.Consumer

class CommandRunner : JPanel() {
    init {

        // components init
        val inputLabel = JLabel("Input:")
        val commandInput = JTextArea()
        val runBtn = JButton("Run")
        val outputLabel = JLabel("Output:")
        val outputDisplay = JTextArea()

        // components config
        commandInput.rows = 3
        outputDisplay.isEditable = false
        runBtn.addActionListener { _ ->
            try {
                runCommand(commandInput.text)
            } catch (e: Exception) {
                outputDisplay.foreground = Color.RED
                outputDisplay.text = e.message
            }
        }

        // layout
        this.layout = MigLayout()
        val components = listOf(
            inputLabel,
            commandInput,
            runBtn,
            outputDisplay,
            outputLabel
        )
        components.forEach(Consumer { cmp -> this.add(cmp, "wrap") })
    }

    private fun runCommand(command: String) {}
}